lib/Optimizer/CodeGen/QuakeToLLVM.cpp

```diff
@@ -1213,6 +1214,137 @@ class RemoveRelaxSizeRewrite : public OpConversionPattern<quake::RelaxSizeOp> {
     return success();
   }
 };
+
+class CustomUnitaryOpRewrite
```

Comment:
Add a comment here that this pattern is reached if and only if the target is a simulator. The QIS functions being generated here are nonsensical on hardware.

We probably want to test that explicitly in fact.

Reply:
Python test present - python/tests/backends/test_Quantinuum_LocalEmulation_kernel.py